txnkv/txnlock/lock_resolver.go

```diff
@@ -266,11 +311,13 @@ func (lr *LockResolver) BatchResolveLocks(bo *retry.Backoffer, locks []*Lock, lo
 		}
 
 		if status.ttl > 0 {
-			logutil.BgLogger().Error("BatchResolveLocks fail to clean locks, this result is not expected!")
-			return false, errors.New("TiDB ask TiKV to rollback locks but it doesn't, the protocol maybe wrong")
+			if checkTxnExpired == math.MaxUint64 {
+				logutil.BgLogger().Error("BatchResolveLocks fail to clean locks, this result is not expected!")
+				return false, errors.New("TiDB ask TiKV to rollback locks but it doesn't, the protocol maybe wrong")
+			}
+		} else {
```

Comment:
If some locks are not expired and not resolved, do you still consider `BatchResolveLocks` as success?

Reply:
Yes, it justly resolves the expired lock and some lock, ts of which is smaller than safepoint.